### PR TITLE
Remove support for *args in pandas wrappers

### PR DESCRIPTION
**Pull request by casperdcl**
Fix total computation for pandas apply

migrated from #244, and rebased

### CODE REVIEW

**Comment by casperdcl** [comment]
@aplavin @CrazyPython not sure what you wanted to achieve with this. seems like once all the commits from #244 were squashed there's just the one change (drop support of `*args`). did i miss something?

File: tqdm/_tqdm.py [hunk 1]
```diff
@@ -390,22 +390,29 @@ def pandas(tclass, *targs, **tkwargs):
...
                 if total is None:  # not grouped
-                    total = len(df) if isinstance(df, Series) \
-                        else df.size // len(df)
+                    if df_function == 'applymap':
+                        total = df.size
+                    else:
+                        axis = kwargs.get('axis', 0)
+                        total = df.shape[axis]
+                        if isinstance(df, DataFrame):
+                            total += 1  # pandas calls update once too many
```
Comment (lrq3000) [inline tqdm/_tqdm.py]: So to be concise: TODO:

- [ ] Add tqdm warning message `if len(args) > 0`
- [ ] (maybe) add unit tests for series and dataframe length detection (according to axis and method used).

Comment (casperdcl) [inline tqdm/_tqdm.py]: I think we need to get someone who develops similar things in pandas to take a look. Don't really have the time to investigate seeing as I don't even use this feature.

**Comment by lrq3000** [comment]
Ok I agree, then let's keep this PR open until we get someone competent to
review and contribute to this.

2016-10-29 23:36 GMT+02:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by CrazyPython** [comment]
@casperdcl huh?

**Comment by lrq3000** [comment]
Casper does not frequently use pandas, he already said so. He just
implemented the tqdm_pandas submodule using only his great insights into
the pandas API. But we are reaching a state where we need someone that is
more expert than us in pandas. I used pandas for a whole project in the
past, so you can say that I am a casual user, and this is getting too
complex for me too.

2016-10-30 0:19 GMT+02:00 CrazyPython <notifications@github.com>:

**Comment by CrazyPython** [comment]
add help-wanted.

@aplavin review this

**Comment by aplavin** [comment]
@CrazyPython well, this is my patch - how can I review it? :) To best of my knowledge it's correct, but I'm not a pandas developer either.

**Comment by lrq3000** [comment]
Thank's @aplavin. Well, then we either should just merge it in and see if it fixes issues (or if someone reports incorrect behavior), or add additional unit tests.

**Comment by casperdcl** [comment]
this fails unit tests on my machine `FAIL: Test pandas.DataFrame[.series].progress_apply` in tests_pandas.py line 60:
```
AssertionError: 
Expected:
100% at least twice
```

**Comment by lrq3000** [comment]
Ok I'll have a look, the tests passed last time I tried.

2016-11-12 17:21 GMT+01:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by lrq3000** [comment]
@casperdcl Ah sorry I confused this PR with #271. I have no idea why this fails, but probably because of the different axis handling. We should put this PR to the background until someone with more experience can provide a better fix...

**Comment by casperdcl** [comment]
from #322:
``` python
# Precompute total iterations
if kwargs.get('axis', 0)==1:
    total = len(df)
        else:
            total = getattr(df, 'ngroups', None)
                if total is None:  # not grouped
                    total = len(df) if isinstance(df, Series) \
                        else df.size // len(df)
                else:
                    total += 1  # pandas calls update once too many
```

**Comment by casperdcl** [comment]
Just rebased onto master. But CI now failing.

**Comment by cancan101** [comment]
any updates on this?

**Comment by casperdcl** [comment]
well if anyone (@cancan101?) would like to fix the conflicts and unit tests, we'd be happy to merge this in...

**Comment by Raj-JainHC** [comment]
@"People with write access", can anyone please review and merge the changes. 🙏

**Comment by chengs** [comment]
I am using pandas frequently. Maybe I could help.

**Comment by casperdcl** [comment]
thanks @chengs

**Comment by chengs** [comment]
even if we exclude args, still there is a problem to get a correct "total". 

pandas apply calls func twice on the first column/row to decide whether it can take a fast or slow code path.

so, sometimes apply_func will run one more time, but maybe not if the func is slow.

still trying to find a way to determine if total = total + 1 or not.

**Comment by chengs** [comment]
please check #521 @casperdcl @lrq3000 
I cannot commit to this #299

**Pull request closed without merging**